bftengine/src/bcstatetransfer/AsyncStateTransferCRE.cpp

```diff
@@ -197,7 +196,8 @@ std::shared_ptr<ClientReconfigurationEngine> CreFactory::create(std::shared_ptr<
   IStateClient* pbc = new PollBasedStateClient(bftClient, cre_config.interval_timeout_ms_, 0, cre_config.id_);
   auto cre =
       std::make_shared<ClientReconfigurationEngine>(cre_config, pbc, std::make_shared<concordMetrics::Aggregator>());
-  cre->registerHandler(std::make_shared<ReplicaTLSKeyExchangeHandler>());
+  if (bftEngine::ReplicaConfig::instance().isReadOnly)
+    cre->registerHandler(std::make_shared<ReplicaTLSKeyExchangeHandler>());
   if (!bftEngine::ReplicaConfig::instance().isReadOnly) cre->registerHandler(std::make_shared<ScalingReplicaHandler>());
```

Comment:
else?

Reply:
we don't need the replicaTLSKeyExchangeHandler in regular replicas.